spec/ibandit/iban_spec.rb

```diff
@@ -353,6 +353,109 @@
                                      branch_code: "is invalid")
       end
     end
+
+    context "when the IBAN was created with local details for New Zealand" do
+      let(:arg) do
+        {
+          country_code: "NZ",
+          bank_code: "11",
+          branch_code: "2222",
+          account_number: account_number,
+        }
+      end
+      context "with a 3 digit account number suffix" do
+        let(:account_number) { "3333333-044" }
+
+        its(:country_code) { is_expected.to eq("NZ") }
+        its(:bank_code) { is_expected.to eq("11") }
+        its(:branch_code) { is_expected.to eq("2222") }
+        its(:account_number) { is_expected.to eq("3333333044") }
+        its(:swift_bank_code) { is_expected.to eq("11") }
+        its(:swift_branch_code) { is_expected.to eq("2222") }
+        its(:swift_account_number) { is_expected.to eq("3333333044") }
+        its(:swift_national_id) { is_expected.to eq("112222") }
+        its(:iban) { is_expected.to be_nil }
+        its(:pseudo_iban) { is_expected.to eq("NZZZ1122223333333044") }
+        its(:valid?) { is_expected.to eq(true) }
+        its(:to_s) { is_expected.to eq("") }
+
+        context "with a wrongly formatted account number" do
+          # 3-digit NZ account number suffixes must always start with 0
+          let(:account_number) { "3333333-944" }
+
+          it "is invalid and has the correct errors" do
+            expect(subject.valid?).to eq(false)
+            expect(subject.errors).to eq(account_number: "is invalid")
+          end
+        end
+      end
+      context "with a 2 digit account number suffix" do
+        let(:account_number) { "3333333-44" }
+
+        its(:country_code) { is_expected.to eq("NZ") }
+        its(:bank_code) { is_expected.to eq("11") }
+        its(:branch_code) { is_expected.to eq("2222") }
+        its(:account_number) { is_expected.to eq("3333333044") }
+        its(:swift_bank_code) { is_expected.to eq("11") }
+        its(:swift_branch_code) { is_expected.to eq("2222") }
+        its(:swift_account_number) { is_expected.to eq("3333333044") }
+        its(:swift_national_id) { is_expected.to eq("112222") }
+        its(:iban) { is_expected.to be_nil }
+        its(:pseudo_iban) { is_expected.to eq("NZZZ1122223333333044") }
+        its(:valid?) { is_expected.to eq(true) }
+        its(:to_s) { is_expected.to eq("") }
+      end
+      context "with bank and branch code embedded in account_number field" do
+        let(:arg) do
+          {
+            country_code: "NZ",
+            account_number: "11-2222-3333333-44",
```

Comment:
We will also want to see some specs confirming if you mix and match both incorrectly (e.g. pass this + a branch code etc), you get a validation error.